### ISSUE TITLE
Task: Make mixed atomic Jetpack e2e reruns use the same variations

### ISSUE THREAD

**Issue by dpasque** [Automattic/wp-calypso#82950]
### Details

Two fixes to make on the build smoke runs:
1. I need to make sure that reruns use the same variation as the first run. 
2. The artifacts have "undefined" in the name on rerun, need to figure out what that is

### Checklist

- [ ]
- [ ]

**Comment by worldomonation** [comment]
For more context on what this describes:

The initial run of a particular spec may be against `wp-beta` and when it fails, you see the following:
```
  Blocks: Media (Upload) (Atomic: wp-beta)
```

and then on the re-run:

```
  Blocks: Media (Upload) (Atomic: default)
```

**Comment by worldomonation** [comment]
I've been thinking about on and off today, and keep coming back to one thing: when tests are triggered via Jest, each test doesn't actually know what environment it's being run against other than what it knows via env variables. 

To address this issue, we almost have to do create an one-to-one mapping of the spec files being run, and the associated environment for the file. Something like this:

- spec_A
  - target: php-old
  - jest worker id: 2
- spec_B
  - target: private
  - jest worker id: 9
- spec_C
  - target: wp-beta
  - jest worker id: 4

and if spec_A and spec_C fail, the mapping knows what target they are supposed to retry against.

The [Jest Environment configuration layer](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/jest-playwright-config/environment.ts) is the place where we can override behavior, but it runs for each instance of a Jest worker. Keeping track of the "mapping" needs to be done at a level higher than that, but I'm not sure if there's such a thing in the Jest environment itself.

**Comment by dpasque** [comment]
Fixed in #83009
